framework/src/outputs/Checkpoint.C

```diff
@@ -72,6 +80,67 @@ Checkpoint::directory() const
   return _file_base + "_" + _suffix;
 }
 
+void
+Checkpoint::outputStep(const ExecFlagType & type)
+{
+  // Output is not allowed
+  if (!_allow_output && type != EXEC_FORCED)
+    return;
+
+  // If recovering disable output of initial condition, it was already output
+  if (type == EXEC_INITIAL && _app.isRecovering())
+    return;
+
+  // Check whether we should output, then do it.
+  if (shouldOutput(type))
+  {
+    TIME_SECTION("outputStep", 2, "Outputting Step");
+    output(type);
+  }
+}
+
+bool
+Checkpoint::shouldOutput(const ExecFlagType & type)
+{
+  switch (AUTOSAVE_TYPE(_is_autosave))
+  {
+    case NONE:
+      // Only call for onInterval now if it isn't an autosave.
+      if (onInterval() || type == EXEC_FINAL)
+        return Output::shouldOutput(type);
+      return false;
+
+    case SYSTEM_AUTOSAVE:
+      // if this is a pure system-created autosave through AutoCheckpointAction,
+      // then sync across processes and only output one time per signal received.
+      comm().max(Moose::autosave_flag);
+      if (Moose::autosave_flag)
+      {
+        Moose::autosave_flag = 0;
```

Comment:
You should probably output to the `_console` here. Something to the effect of "Unix Signal SIGUSR1 caught, saving Checkpoint". Note, you'll need to do that in two places with the duplication of code.